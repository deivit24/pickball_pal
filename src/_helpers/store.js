import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
