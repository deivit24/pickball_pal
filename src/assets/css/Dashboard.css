.no-profile {
  text-align: center;
  padding: 30px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  box-shadow: 0 0 8px 2px #afafaf;
  border-radius: 25px;
}
.yes-profile {
  padding: 25px;
  display: flex;
  flex-direction: column;
  justify-content: center;

  box-shadow: 0 0 8px 2px #afafaf;
  border-radius: 25px;
}

.no-profile h2 {
  margin-bottom: 25px;
}

.no-profile img {
  width: 100%;
}

p {
  font-size: 1.2rem;
}
