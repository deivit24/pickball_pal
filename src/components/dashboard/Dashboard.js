// React
import React, { useEffect } from 'react';
// React Router Dom
// import { Redirect, Link } from 'react-router-dom';
// Redux
import { useDispatch, useSelector } from 'react-redux';
// Get Current Profile and Get Messages actions
import { getCurrentProfile, getMessages } from '../../_actions/profile';
// Loading
import Loading from '../layout/Loading';
// Dashboard Components
import DashboardCard from './DashboardCard';
import DashboardActions from './DashboardActions';
import DashboardNav from './DashboardNav';

const Dashboard = () => {
  const auth = useSelector((store) => store.auth);
  const meProfile = useSelector((store) => store.profile);

  const dispatch = useDispatch();
  const { profile, loading } = meProfile;
  const { user } = auth;

  useEffect(() => {
    dispatch(getCurrentProfile());
    dispatch(getMessages());
    document.title = ` ${user && user.first_name}'s Dashboard`;
  }, [dispatch, user]);

  let result;

  let name =
    user &&
    user.first_name.toUpperCase() +
      ' ' +
      (user && user.last_name.toUpperCase());

  if (loading && profile === null) {
    result = <Loading type="spin" color="#3f50b5" />;
  } else {
    result = (
      <>
        <div id="Dashboard" className="container mt-5">
          <div className="row">
            <div className="col-md-4">
              <DashboardCard name={name} avatar={user && user.avatar} />
              {profile !== null ? <DashboardActions /> : ''}
            </div>

            <div className="col-md-8">
              <DashboardNav />
            </div>
          </div>
        </div>
      </>
    );
  }

  return <>{result}</>;
};

export default Dashboard;
