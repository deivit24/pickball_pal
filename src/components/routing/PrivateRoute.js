// React
import React from 'react';
// React Router DOM
import { Route, Redirect } from 'react-router-dom';
// Redux
import { useSelector } from 'react-redux';

const PrivateRoute = ({ exact, path, children }) => {
  const auth = useSelector((store) => store.auth);

  const { isAuth, loading } = auth;

  if (!isAuth && !loading) {
    return <Redirect to="/login" />;
  }

  return (
    <Route exact={exact} path={path}>
      {children}
    </Route>
  );
};

export default PrivateRoute;
