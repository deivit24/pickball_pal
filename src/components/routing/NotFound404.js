import React from 'react';
import '../../assets/css/404.css';

const NotFound404 = () => {
  return (
    <section className="container NotFound">
      <h1>Sorry, couldn't find what you were looking for</h1>
      <h3>Error 404</h3>
      <img
        src="https://img.pngio.com/bicycle-boy-by-tanka-dribbble-dribbble-bicycle-png-gif-800_600.gif"
        alt=""
      />
    </section>
  );
};

export default NotFound404;
