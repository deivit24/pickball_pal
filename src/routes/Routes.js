// React
import React from 'react';
// React Route DOM
import { Switch, Route } from 'react-router-dom';
// Components for Routes
import Login from '../components/auth/Login';
import Register from '../components/auth/Register';
import Dashboard from '../components/dashboard/Dashboard';
import Profiles from '../components/profiles/Profiles';
import Profile from '../components/profile/Profile';
import Home from '../components/layout/Home';
import PrivateRoute from '../components/routing/PrivateRoute';
import NotFound404 from '../components/routing/NotFound404';
import CreateProfile from '../components/forms/CreateProfile';
import EditProfile from '../components/forms/EditProfile';
import UploadPhoto from '../components/forms/UploadPhoto';
import AddPlaces from '../components/forms/AddPlaces';
import Posts from '../components/posts/Posts';
import Post from '../components/post/Post';

const Routes = () => {
  return (
    <Switch>
      <Route exact path="/">
        <Home />
      </Route>
      <Route exact path="/login">
        <Login />
      </Route>
      <Route exact path="/register">
        <Register />
      </Route>
      <Route exact path="/profiles">
        <Profiles />
      </Route>
      <Route exact path="/profile/:id">
        <Profile />
      </Route>
      <PrivateRoute exact path="/dashboard">
        <Dashboard />
      </PrivateRoute>
      <PrivateRoute exact path="/create-profile">
        <CreateProfile />
      </PrivateRoute>
      <PrivateRoute exact path="/edit-profile">
        <EditProfile />
      </PrivateRoute>
      <PrivateRoute exact path="/upload-photo">
        <UploadPhoto />
      </PrivateRoute>
      <PrivateRoute exact path="/add-places">
        <AddPlaces />
      </PrivateRoute>
      <PrivateRoute exact path="/posts">
        <Posts />
      </PrivateRoute>
      <PrivateRoute exact path="/posts/:id">
        <Post />
      </PrivateRoute>
      <Route>
        <NotFound404 />
      </Route>
    </Switch>
  );
};

export default Routes;
