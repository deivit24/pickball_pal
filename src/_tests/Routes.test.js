import React from 'react';
import { render } from '@testing-library/react';
import PrivateRoute from '../components/routing/PrivateRoute';
import NotFound404 from '../components/routing/NotFound404';
import Dashboard from '../components/dashboard/Dashboard';
import Routes from '../routes/Routes';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../_helpers/store';

//Testing Private Routes

describe('Testing the Private Routes', () => {
  it('it renders Private Routes without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <PrivateRoute exact path="/">
            <Dashboard />
          </PrivateRoute>
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <PrivateRoute exact path="/">
            <Dashboard />
          </PrivateRoute>
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

// Testing Not Found Routes

describe('Testing the Not Found Routes', () => {
  it('it renders Not Found Routes without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <NotFound404 />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <NotFound404 />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

// Testing Routes

describe('Testing the Routes', () => {
  it('it renders Routes without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <Routes />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Routes />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
