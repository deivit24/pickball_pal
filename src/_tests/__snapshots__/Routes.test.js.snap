// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Not Found Routes matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="container NotFound"
  >
    <h1>
      Sorry, couldn't find what you were looking for
    </h1>
    <h3>
      Error 404
    </h3>
    <img
      alt=""
      src="https://img.pngio.com/bicycle-boy-by-tanka-dribbble-dribbble-bicycle-png-gif-800_600.gif"
    />
  </section>
</DocumentFragment>
`;

exports[`Testing the Private Routes matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container Loading"
  >
    <div
      style="fill: #3f50b5; height: 100px; width: 100px;"
    >
      <svg
        viewBox="0 0 32 32"
        xmlns="http://www.w3.org/2000/svg"
      >
        
  
        <path
          d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4"
          opacity=".25"
        />
        
  
        <path
          d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z"
        >
          
    
          <animatetransform
            attributeName="transform"
            dur="0.8s"
            from="0 16 16"
            repeatCount="indefinite"
            to="360 16 16"
            type="rotate"
          />
          
  
        </path>
        

      </svg>
      

    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the Routes matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="px-0 container-fluid"
    id="Home"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-12 text-center"
        >
          <h1>
            Welcome to Pickleball Pal
          </h1>
          <h3
            class="mb-3"
          >
            Find local pickleball players near you
          </h3>
          <div
            class=" mb-4"
          >
            <form
              autocomplete="on"
              id="Search"
              novalidate=""
            >
              <div
                class="searchIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <div
                class="MuiInputBase-root input-search"
              >
                <input
                  class="MuiInputBase-input"
                  name="search"
                  placeholder="Enter City or Zip Code"
                  type="text"
                  value=""
                />
              </div>
            </form>
            <small
              style="color: white;"
            >
              Press Enter (eg. Austin, TX or 78745)
            </small>
          </div>
          <a
            class="mr-3"
            href="/register"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Register
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </a>
          <a
            href="/profiles"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                View Players
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container"
    id="HomeCards"
  >
    <h1>
      Perks of signing up
    </h1>
    <div
      class="MuiGrid-root grid MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root paper MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt=""
            src="person.jpg"
          />
          <h4>
            Create a Profile
          </h4>
          <p>
            You can create and edit a profile that other players can find. You can display your bio, playing style level, social media accounts, and where you play.
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root paper MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt=""
            src="message.jpg"
          />
          <h4>
            Direct Message Players
          </h4>
          <p>
            You can send other players direct messages and directly set up your own match or practice schedule with the player. We take no payment for connecting players. 
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root paper MuiPaper-elevation1 MuiPaper-rounded"
        >
          <img
            alt=""
            src="earth.jpg"
          />
          <h4>
            Post to the Community
          </h4>
          <p>
            You can post to the Pickleball community and filter posts by location. That way your post doesn't get lost in the world. Comment and like other posts as well. 
          </p>
        </div>
      </div>
    </div>
    <a
      href="/register"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Create Profile
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </a>
  </div>
  <div
    id="HomeMission"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-6"
        >
          <img
            alt=""
            src="pickleball.jpg"
          />
        </div>
        <div
          class="col-md-6"
        >
          <h1>
            We're on a mission
          </h1>
          <p>
            Our goal is to grow Pickleball by making it easier to connect with players, get practice, and play matches at your local courts
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    id="HomeFooter"
  >
    <div
      class="container"
    >
      <h1>
        Created by 
        <a
          href="https://deivitsalazar.com"
          target="a_blank"
        >
           David Salazar
        </a>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
